crates/re_renderer/examples/picking.rs

```diff
@@ -0,0 +1,194 @@
+    }
+
+    fn on_cursor_moved(&mut self, position_in_pixel: glam::UVec2) {
+        self.picking_position = position_in_pixel;
+    }
+
+    fn new(_re_ctx: &mut re_renderer::RenderContext) -> Self {
+        let mut rnd = <rand::rngs::StdRng as rand::SeedableRng>::seed_from_u64(42);
+        let random_point_range = -5.0_f32..5.0_f32;
+        let point_count = 10000;
+
+        // Split point cloud into several batches to test picking of multiple objects.
+        let point_sets = (0..2)
+            .map(|_| PointSet {
+                positions: (0..point_count)
+                    .map(|_| {
+                        glam::vec3(
+                            rnd.gen_range(random_point_range.clone()),
+                            rnd.gen_range(random_point_range.clone()),
+                            rnd.gen_range(random_point_range.clone()),
+                        )
+                    })
+                    .collect_vec(),
+                radii: std::iter::repeat(Size::new_scene(0.08))
+                    .take(point_count)
+                    .collect_vec(),
+                colors: (0..point_count)
+                    .map(|_| random_color(&mut rnd))
+                    .collect_vec(),
+                picking_ids: (0..point_count as u64)
+                    .map(PickingLayerInstanceId)
+                    .collect_vec(),
+            })
+            .collect_vec();
+
+        Picking {
+            point_sets,
+            scheduled_picking_rects: HashMap::default(),
+            picking_position: glam::UVec2::ZERO,
+        }
+    }
+
+    fn draw(
+        &mut self,
+        re_ctx: &mut re_renderer::RenderContext,
+        resolution: [u32; 2],
+        time: &framework::Time,
+        pixels_from_point: f32,
+    ) -> Vec<framework::ViewDrawResult> {
+        self.handle_incoming_picking_data(re_ctx, time.seconds_since_startup());
+
+        let mut view_builder = ViewBuilder::default();
+
+        // TODO(#1426): unify camera logic between examples.
+        let camera_position = glam::vec3(1.0, 3.5, 7.0);
+
+        view_builder
+            .setup_view(
+                re_ctx,
+                TargetConfiguration {
+                    name: "OutlinesDemo".into(),
+                    resolution_in_pixel: resolution,
+                    view_from_world: macaw::IsoTransform::look_at_rh(
+                        camera_position,
+                        glam::Vec3::ZERO,
+                        glam::Vec3::Y,
+                    )
+                    .unwrap(),
+                    projection_from_view: Projection::Perspective {
+                        vertical_fov: 70.0 * std::f32::consts::TAU / 360.0,
+                        near_plane_distance: 0.01,
+                    },
+                    pixels_from_point,
+                    outline_config: None,
+                    ..Default::default()
+                },
+            )
+            .unwrap();
+
+        let picking_rect_size = 32;
```

Comment:
I would like it to be a power of two because gpus are notoriously bad with non-power of two textures still. But, no it's not a hard restriction. I'll try with 31 and see if that yields more accurate picking here